File: src/TmrOverlay.App/Overlays/Flags/FlagsOverlayDefinition.cs
using System.Drawing;
using TmrOverlay.Core.Overlays;
using TmrOverlay.Core.Settings;

namespace TmrOverlay.App.Overlays.Flags;

internal static class FlagsOverlayDefinition
{
    public const string PrimaryScreenDefaultId = "primary-screen-default";
    public const int MinimumWidth = 180;
    public const int MaximumWidth = 960;
    public const int MinimumHeight = 96;
    public const int MaximumHeight = 420;

    public static OverlayDefinition Definition { get; } = new(
        Id: "flags",
        DisplayName: "Flags",
        DefaultWidth: 360,
        DefaultHeight: 170,
        ShowSessionFilters: false,
        ShowScaleControl: true,
        ShowOpacityControl: false,
        FadeWhenLiveTelemetryUnavailable: true);

    public static Size ResolveSize(OverlaySettings settings)
    {
        return new Size(
            Math.Clamp(settings.Width > 0 ? settings.Width : Definition.DefaultWidth, MinimumWidth, MaximumWidth),
            Math.Clamp(settings.Height > 0 ? settings.Height : Definition.DefaultHeight, MinimumHeight, MaximumHeight));
    }
}

File: local-mac/TmrOverlayMac/Sources/TmrOverlayMac/Overlays/Flags/FlagsOverlayDefinition.swift
import AppKit

enum FlagsOverlayDefinition {
    static let primaryScreenDefaultId = "primary-screen-default"
    static let minimumWidth = 180.0
    static let maximumWidth = 960.0
    static let minimumHeight = 96.0
    static let maximumHeight = 420.0

    static let definition = OverlayDefinition(
        id: "flags",
        displayName: "Flags",
        defaultSize: NSSize(width: 360, height: 170),
        showSessionFilters: false,
        showScaleControl: false,
        showOpacityControl: false,
        fadeWhenLiveTelemetryUnavailable: true
    )

    static func resolveSize(_ settings: OverlaySettings) -> NSSize {
        NSSize(
            width: min(max(settings.width > 0 ? settings.width : definition.defaultSize.width, minimumWidth), maximumWidth),
            height: min(max(settings.height > 0 ? settings.height : definition.defaultSize.height, minimumHeight), maximumHeight)
        )
    }
}
